src/pages/library/datasetBuilder/dataset-builder-types.ts

```diff
@@ -27,7 +27,7 @@ export interface ProgramDataRangeCriteria extends Criteria {
 export interface ProgramDataListCriteria extends Criteria {
   kind: 'list';
   listOption: ProgramDataListOption;
-  value: ProgramDataListValueOption;
+  valuesSelected: ProgramDataListValueOption[];
```

Comment:
I think selection is a concept here, but not in the Ajax response version (they're separate) - I think I got a bit confused that this wasn't all the values possible in the list, but rather a selection of the values possible (at least in our implementation)

Reply:
> Actually, isn't listOption.values() the full list of objects? So the diff of listOption.values() - valuesSelected would be unselected values?

It's true that the list of all possible values is available indirectly in the criteria, but that is (IMO) incidental to how we represent list value options, not part of the criteria itself. They are selected in the UI, but when they become part of the criteria definition they are no longer "selected". I'm thinking about this as input for a service API that's separate from the UI that builds it.

On the other hand, this could be thought of as representing the UI state only and we'd have a transform that produces the API version of it.

Reply:
My understanding is that this was `ProgramDataListOption` in the DatasetBuilder ajax code, as a separate object. And that this represents UI state only because it didn't live in the ajax.